src/boot/efi/secure-boot.c

```diff
@@ -103,6 +139,23 @@ EFI_STATUS secure_boot_enroll_at(EFI_FILE *root_dir, const char16_t *path, bool
                         log_error_status(err, "Failed reading file %ls\\%ls: %m", path, sb_vars[i].filename);
                         goto out_deallocate;
                 }
+                if (strcmp(sb_vars[i].name, u"PK") == 0 && sb_vars[i].size > 20) {
+                        uint32_t *sigsize = (uint32_t*)(sb_vars[i].buffer + 16);
```

Comment:
a few magic numbers in this block, at least a comment explaining what they are would be good